R/utils.R

```diff
@@ -278,27 +271,68 @@ configure_spark <- function(call = caller_env()) {
 }
 
 locate_install_unixodbc <- function() {
-  tryCatch(
-    {
-      unixodbc_prefix <- system("odbc_config --lib-prefix", intern = TRUE)
-      return(paste0(unixodbc_prefix, "/libodbcinst.dylib"))
-    },
-    error = function(e) {}
+  libodbcinst <- c(
+    system_safely("odbc_config --lib-prefix"),
+    system_safely("pkg-config --variable=libdir odbc")
   )
 
+  if (length(libodbcinst) > 0) {
+    return(libodbcinst[1])
+  }
+
   common_dirs <- c(
+    "/usr/lib",
+    "/usr/local",
+    "/lib",
     "/usr/local/lib",
     "/opt/homebrew/lib",
     "/opt/homebrew/opt/unixodbc/lib"
   )
 
   list.files(
     common_dirs,
-    pattern = "libodbcinst\\.dylib$",
+    pattern = "libodbcinst\\.(dylib|so)$",
     full.names = TRUE
   )
 }
 
+system_safely <- function(x) {
+  tryCatch(
+    {
+      unixodbc_prefix <- system(x, intern = TRUE, ignore.stderr = TRUE)
+      return(paste0(unixodbc_prefix, "/libodbcinst.", libodbcinst_extension()))
+    },
+    error = function(e) {},
+    warning = function(w) {}
+  )
+  character()
+}
+
+libodbcinst_extension <- function() {
```

Comment:
Might be a little nicer to make this `libodbcinst_filename`? (e.g return `libodbcinst.dylib`)